src/core/core.controller.js

```diff
@@ -342,11 +321,15 @@ helpers.extend(Chart.prototype, /** @lends Chart */ {
 
 		if (options.scales) {
 			items = items.concat(
-				(options.scales.xAxes || []).map(function(xAxisOptions) {
-					return {options: xAxisOptions, dtype: 'category', dposition: 'bottom'};
-				}),
-				(options.scales.yAxes || []).map(function(yAxisOptions) {
-					return {options: yAxisOptions, dtype: 'linear', dposition: 'left'};
+				Object.entries(options.scales).map(function(entry) {
+					var axisID = entry[0];
+					var axisOptions = entry[1];
+					var isHorizontal = axisID.charAt(0).toLowerCase() === 'x';
+					return {
```

Comment:
If `entry === 'r'` maybe we should return a different object?

Reply:
Right, can remove that `options.scale` from there  too. Updated.